plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,471 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package fritzbox
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/logger"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/tdrn-org/go-tr064"
+	"github.com/tdrn-org/go-tr064/mesh"
+	"github.com/tdrn-org/go-tr064/services/igddesc/igdicfg"
+	"github.com/tdrn-org/go-tr064/services/tr64desc/deviceinfo"
+	"github.com/tdrn-org/go-tr064/services/tr64desc/hosts"
+	"github.com/tdrn-org/go-tr064/services/tr64desc/wancommonifconfig"
+	"github.com/tdrn-org/go-tr064/services/tr64desc/wandslifconfig"
+	"github.com/tdrn-org/go-tr064/services/tr64desc/wanpppconn"
+	"github.com/tdrn-org/go-tr064/services/tr64desc/wlanconfig"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const pluginName = "fritzbox"
+
+const defaultTimeout = config.Duration(10 * time.Second)
+
+type Fritzbox struct {
+	Devices           []string        `toml:"devices"`
+	DeviceInfo        bool            `toml:"device_info"`
+	WanInfo           bool            `toml:"wan_info"`
+	PppInfo           bool            `toml:"ppp_info"`
+	DslInfo           bool            `toml:"dsl_info"`
+	WlanInfo          bool            `toml:"wlan_info"`
+	HostsInfo         bool            `toml:"hosts_info"`
+	FullQueryCycle    int             `toml:"full_query_cycle"`
+	Timeout           config.Duration `toml:"timeout"`
+	TlsSkipVerify     bool            `toml:"tls_skip_verify"`
+	Debug             bool            `toml:"debug"`
```

Comment:
No need to have a plugin specific debug option, the log.Debug() methods can be used for that.

Reply:
Will remove

Reply:
Done via ccd459aee13ea59a56e50afcc9f0e590f6967638